Net/SMTP.php

```diff
@@ -623,23 +623,69 @@ public function starttls()
                     /* STREAM_CRYPTO_METHOD_TLS_ANY_CLIENT constant does not exist
                      * and STREAM_CRYPTO_METHOD_SSLv23_CLIENT constant is
                      * inconsistent across PHP versions. */
-                    $crypto_method = STREAM_CRYPTO_METHOD_TLS_CLIENT
-                    | @STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT
-                    | @STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
+                    $crypto_method = STREAM_CRYPTO_METHOD_TLS_CLIENT;
+
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT;
+                    }
+
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
+                    }
+
+                    //New Crypto-Method since PHP7.4
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT;
+                    }
                 }
-                if (PEAR::isError($result = $this->socket->enableCrypto(true, $crypto_method))) {
+
+                for ($attempts = 1; $attempts < 100; $attempts++) {
+                    $result = $this->socket->enableCrypto(true, $crypto_method);
```

Comment:
Do we need to check for a `PEAR::isError($result)` result from this method call?

Reply:
There is a check for `PEAR::isError($result)` at line 651. Do you think a sooner check is necessary?

Reply:
Yes, but it is also possible for these inner (additional) checks to fail, too. I don't know if that should be considered a fatal error or now, though.